File: src/main/java/model/Genius.java
package model;

import java.util.ArrayList;
import java.util.List;
import jakarta.xml.bind.annotation.XmlElement;
import jakarta.xml.bind.annotation.XmlRootElement;

@XmlRootElement
public class Genius {

    private Type type;
    private final List<Integer> colors = new ArrayList<>();
    private int index = 0;

    public Genius() {
        colors.add((int) Math.floor(Math.random() * 4));
    }

    public void checkColor(int color) {
        if (color != colors.get(index)) {
            colors.clear();
            index = 0;
            type = Type.WRONG_COLOR;
        } else {
            if (colors.size() == ++index) {
                index = 0;
                colors.add((int) Math.floor(Math.random() * 4));
                type = Type.NEW_COLOR;
            } else {
                type = Type.CORRECT_COLOR;
            }
        }
    }

    @XmlElement
    public List<Integer> getColors() {
        return colors;
    }

    @XmlElement
    public Type getType() {
        return type;
    }

    @XmlElement
    public int getIndex() {
        return index;
    }
}

File: docs/scripts/Genius.js
import Type from './Type.js';

export default class Genius {
    constructor() {
        this.type = null;
        this.colors = [];
        this.colors.push(Math.floor(Math.random() * 4));
        this.index = 0;
    }
    checkColor(color) {
        if (color !== this.colors[this.index]) {
            this.colors = [];
            this.index = 0;
            this.type = Type.WRONG_COLOR;
        } else {
            if (this.colors.length === ++this.index) {
                this.index = 0;
                this.colors.push(Math.floor(Math.random() * 4));
                this.type = Type.NEW_COLOR;
            } else {
                this.type = Type.CORRECT_COLOR;
            }
        }
    }
    getType() {
        return this.type;
    }
    getColors() {
        return this.colors;
    }
    getIndex() {
        return this.index;
    }
}
